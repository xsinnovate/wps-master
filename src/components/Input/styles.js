import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  label: {
    paddingBottom: 4,
  },
  labelText: {
    color: '#777',
  },
  input: {
    height: 40,
    borderColor: '#ccc',
    borderRadius: 5,
    borderWidth: 1,
    paddingHorizontal: 10,
  },
});
