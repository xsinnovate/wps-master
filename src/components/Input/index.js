import PropTypes from 'prop-types';
import React from 'react';
import { TextInput, Text, View } from 'react-native';
import styles from './styles';

export default function Input({
  label,
  onChange,
  value,
  ...rest
}) {
  return (
    <View>
      <View style={styles.label}>
        <Text style={styles.labelText}>
          {label}
        </Text>
      </View>
      <View>
        <TextInput
          {...rest}
          onChangeText={onChange}
          value={value}
          style={styles.input}
        />
      </View>
    </View>
  );
}

Input.propTypes = {
  label: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
};
