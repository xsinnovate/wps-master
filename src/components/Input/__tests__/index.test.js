import React from 'react';
import renderer from 'react-test-renderer';
import { TextInput } from 'react-native';
import Input from '..';

const noop = () => {};

describe('<Input />', () => {
  it('renders correctly', () => {
    const tree = renderer.create((
      <Input
        label="Label"
        onChange={noop}
        value="value"
      />
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should call onChange prop with new text values', () => {
    const spy = jest.fn();
    const instance = renderer.create((
      <Input
        label="Label"
        onChange={spy}
        value="value"
      />
    )).root;

    instance.findByType(TextInput).props.onChangeText('newValue');

    expect(spy).toBeCalledWith('newValue');
  });
});
