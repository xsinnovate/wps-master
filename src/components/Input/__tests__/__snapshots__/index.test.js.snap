// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "paddingBottom": 4,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#777",
        }
      }
    >
      Label
    </Text>
  </View>
  <View>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      style={
        Object {
          "borderColor": "#ccc",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 40,
          "paddingHorizontal": 10,
        }
      }
      underlineColorAndroid="transparent"
      value="value"
    />
  </View>
</View>
`;
